esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -1841,8 +1841,12 @@ int EspHttpBinding::getServiceWsdlOrXsd(IEspContext &context, CHttpRequest* requ
     }
 
     StringBuffer schema;
-    getServiceSchema(context, request, serviceQName, methodQName,
-                     version, isWsdl, false, schema);
+    // Allow derived classes such as WsEcl to use custom getSchema() logic when there is no serviceXmlFilename
+    StringBuffer serviceXmlFilename;
```

Comment:
Here is a concern I have. This does a copy of the serviceXmlFile name, and if present, calls the getServiceSchema method. That method does another copy of the service XML filename. Perhaps there could be a way to avoid two copies and add a "hasSerivceXML" type method. I don't think it practical to pass the service XML filename as a parameter.

If you feel this is valid, let's open a Jira to make that change and add this one as a dependency. It can be done afterwards. It would require a change to HIDL probably. I don't mind making the change. 

As part of that change, the XML filename could be cached, but I don't think that would gain too much.